modules/affile/file-signals.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022 Shikhar Vashistha
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef FILE_SIGNALS_H_INCLUDED
+#define FILE_SIGNALS_H_INCLUDED
+
+#include "signal-slot-connector/signal-slot-connector.h"
+
+typedef struct _FileFlushSignalData FileFlushSignalData;
+typedef struct _FileReopener FileReopener;
+
+struct _FileFlushSignalData
+{
+  const gchar *filename;
+  FileReopener *reopener;
```

Comment:
I think we could allow the file opener plugin to indicate that a reopen is needed via a return value, so we wouldn't have to push the FileReopener instance down to the guts of the rotation plugin.

This way the reopen would  either be delegated to the LogProtoFileWriter or even better back to the driver via the use of the log_pipe_notify() mechanism.

Reply:
Do you mean that we can have a returning flag of some sort for the purpose of reopening the file ?